health/health.d/wmi.conf

```diff
@@ -137,3 +137,175 @@ component: Disk
     delay: down 15m multiplier 1.5 max 1h
      info: disk space utilization
        to: sysadmin
+
+### TCP
+# Source: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/default-dynamic-port-range-tcpip-chang
+# Windows by default allows 16383 (Windows 2008 or newer) and 3976 (old versions) parallel ports.
+#
+# We are adopting conservative values.
+#
+ template: wmi_concurrent_tcp
+       on: wmi.tcp_conns_active
+    class: TCP
+     type: Windows
+component: IIS
+       os: *
+    hosts: *
+     calc: $ipv4 + $ipv6
+    units: connections
+    every: 10s
+     warn: $this > (($status >= $WARNING)  ? (2783) : (2385))
+     crit: $this > (($status == $CRITICAL) ? (3578) : (3180))
+     info: Number of ports used is reaching levels that won't accept more connections.
+       to: sysadmin
+
+ template: wmi_1m_ip_tcp_resets
+       on: wmi.tcp_conns_resets
+    class: Errors
+     type: Windows
+component: Network
+       os: *
+    hosts: *
+   lookup: average -1m at -10s unaligned absolute of ipv4,ipv6
+    units: tcp resets/s
+    every: 10s
+     info: average number of TCP RESETS over the last minute
+
+ template: wmi_10s_ip_tcp_resets
+       on: wmi.tcp_conns_active
+    class: Errors
+     type: System
+component: Network
+       os: *
+    hosts: *
+   lookup: average -10s unaligned absolute of ipv4,ipv6
+    units: tcp resets/s
+    every: 10s
+     warn: $netdata.uptime.uptime > (1 * 60) AND $this > ((($wmi_1m_ip_tcp_resets < 5)?(5):($wmi_1m_ip_tcp_resets)) * (($status >= $WARNING)  ? (1) : (10)))
+    delay: up 20s down 60m multiplier 1.2 max 2h
+  options: no-clear-notification
+     info: average number of sent TCP RESETS over the last 10 seconds. \
+           This can indicate a port scan, \
+           or that a service running on this host has crashed. \
+           Netdata will not send a clear notification for this alarm.
+       to: sysadmin
+
+
+### IIS
+# Source: https://learn.microsoft.com/en-us/answers/questions/495615/how-to-set-to-allow-maximumtcp-connection-in-windo.html
+# We have an alert to notify when TCP ports are exausted, but conggestion can conduct serves for another issue.
+ template: wmi_iis_congestion
+       on: wmi.iis_website_active_connections_count
+    class: Utilization
+     type: Windows
+component: IIS
+       os: *
+    hosts: *
+   lookup: sum -10s of active
```

Comment:
Why sum? Did you mean average?

Reply:
The sample too small to have average, I extended period of time and I made other small adjusts on this PR to use average.

Reply:
The value is a gauge (absolute) - using `sum` is just wrong, it has nothing to do with "sample too small".

Reply:
ok, I already modified.